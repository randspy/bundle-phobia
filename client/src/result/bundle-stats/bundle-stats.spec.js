import React from 'react';
import { shallow } from 'enzyme';
import BundleStats from './bundle-stats';

describe('BundleStats', () => {
  test('should wire component', () => {
    const wrapper = shallow(<BundleStats />);
    expect(wrapper.exists()).toBe(true);
  });

  test('should display minified size in KBs', () => {
    const wrapper = shallow(<BundleStats minified={99 * 1024} />);
    expect(wrapper.find('div.bundle-stats--minified').text()).toEqual('99kB');
  });

  test('should display gzip size in KBs', () => {
    const wrapper = shallow(<BundleStats gzip={88 * 1024} />);
    expect(wrapper.find('div.bundle-stats--gzip').text()).toEqual('88kB');
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <BundleStats minified={99 * 1024} gzip={88 * 1024} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
