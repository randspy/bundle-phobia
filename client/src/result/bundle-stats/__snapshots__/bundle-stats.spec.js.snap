// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BundleStats should render correctly 1`] = `
<div
  className="bundle-stats"
>
  <div
    className="bundle-stats--label bundle-stats--center"
  >
    BUNDLE SIZE
  </div>
  <div
    className="bundle-stats--size"
  >
    <div
      className="bundle-stats--group"
    >
      <div
        className="bundle-stats--minified"
      >
        <div
          className="bundle-stats--number"
        >
          99
        </div>
        <div
          className="bundle-stats--unit"
        >
          kB
        </div>
      </div>
      <div
        className="bundle-stats--center"
      >
        Minified
      </div>
    </div>
    <div
      className="bundle-stats--group"
    >
      <div
        className="bundle-stats--gzip"
      >
        <div
          className="bundle-stats--number"
        >
          88
        </div>
        <div
          className="bundle-stats--unit"
        >
          kB
        </div>
      </div>
      <div
        className="bundle-stats--center"
      >
        Minified + Gzipped
      </div>
    </div>
  </div>
</div>
`;
